File: Greedy/53_maximum_subarray/53.rs
fn max_sub_array(nums: Vec<i32>) -> i32 {
    let mut current_sum: i32 = nums[0];
    let mut max_sum: i32 = nums[0];
    for x in nums[1..].iter() {
        if current_sum < 0 {
            current_sum = *x;
        } else {
            current_sum += *x;
        }
        if current_sum > max_sum {
            max_sum = current_sum;
        }
    }
    return max_sum;
}

File: Greedy/53_maximum_subarray/53.py

def maxSubArray(nums):
        
    current_sum = nums[0]
    max_sum = nums[0]
    
    for i in range(1, len(nums)):
        if current_sum < 0:
            current_sum = nums[i]
        else:
            current_sum += nums[i]
        
        if max_sum < current_sum:
            max_sum = current_sum
        
    return max_sum